app/models/miq_worker/container_common.rb

```diff
@@ -46,6 +69,10 @@ def resource_constraints
 
       mem_limit = self.class.worker_settings[:memory_threshold]
       cpu_limit = self.class.worker_settings[:cpu_threshold_percent]
+
+      # TODO: If request > limit, set request = limit or raise an error
+      # kubeclient will raise each time if we try:
+      # [Kubeclient::HttpError]: Deployment.apps "1-schedule" is invalid: spec.template.spec.containers[0].resources.requests: Invalid value: "567Mi": must be less than or equal to memory limit
```

Comment:
I wonder if we can enforce this via our Settings validation.  That is, if someone tries to change to invalid values in the settings, we raise an error at that time.  We can still do this enforcement here as a safety net, but from a UX perspective, it's better to catch it earlier and don't allow the user to shoot themselves in the foot.

Reply:
Yes, that's a good point.  I was thinking that this was a last ditch effort.  Catching it in validation would prevent:
1) constant need to patch the deployment and repeated logged errors when it fails with invalid values
or 
2) unknown changes to the user's desired values (if we automatically set request to limit if request > limit)

Reply:
I'll look at doing the Settings validation as a follow up PR.   Additionally, should I set `request` == `limit` if `request > limit`?  We can discuss in a follow up PR if so, since this isn't a "new problem", only something I discovered while testing.

Reply:
Opened https://github.com/ManageIQ/manageiq/pull/20889 to validate the workers section of Settings so limits > requests can't be saved.

Reply:
Cool...then I think you can raise an exception here as a safety net (or just let Kube complain directly as it does now)

Reply:
Yeah, good idea.  I think it makes more sense to raise an `ArgumentError` here than to let it complain.

Reply:
ok, pushed changes raising ArgumentError.